dotcom-rendering/src/model/groupTrailsByDates.ts

```diff
@@ -0,0 +1,207 @@
+import type { DCRFrontCard } from '../types/front';
+
+/**
+ * The number of trails per day required (on average) for trails to be
+ * grouped by day instead of month.
+ */
+const MinimumPerDayPopOutFrequency = 2;
+
+/**
+ * Helper type to give easy access to a javascript Date
+ * class representing the date of publication for a trail
+ */
+interface TrailAndDate {
+	trail: DCRFrontCard;
+	date: Date;
+}
+
+/**
+ * Represents a set of trails grouped by their year, month & optionally day of publication.
+ */
+export interface GroupedTrails {
+	year: number;
+	month: number;
+	day: number | undefined;
+	trails: DCRFrontCard[];
+}
+
+/**
+ * Calculates the average number of trails per day across a set of days
+ */
+const getMeanDayFrequency = (
+	days: Array<{ day: number; trails: TrailAndDate[] }>,
+) => {
+	const totalTrails = days
+		.flatMap(({ trails }) => trails.length)
+		.reduce((acc, curr) => acc + curr, 0);
+	return totalTrails / days.length;
+};
+
+/**
+ * Takes a set of trails & returns them grouped by year of publication
+ */
+const groupByYear = (trails: TrailAndDate[]) => {
+	const trailsByYear: Array<{ year: number; trails: TrailAndDate[] }> = [];
+
+	trails.forEach(({ trail, date }) => {
+		const existingYear = trailsByYear.find(
+			({ year }) => year === date.getFullYear(),
```

Comment:
You’re using `getUTCDate`, but not `getUTCFullYear` or `getUTCMonth`. Which should it be?